### PR TITLE
Added a command-line interface.

### PR DESCRIPTION
**Pull request by liautaud**
This PR adds a simple command-line utility which can be used to:

- Compare the output of tfdeploy and tensorflow on a model:
  `cargo run --features tensorflow -- ../tests/plus3.pb -s 1x1xf32 compare`

- Profile the execution of tfdeploy on a model:
  `cargo run -- ../tests/plus3.pb -s 1x1xf32 profile`

### CODE REVIEW

File: cli/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+tfdeploy = { path = ".." }
+conform = { path = "../conform" }
+clap = "2.31"
+log = "0.4"
+error-chain = "0"
+simplelog = "0.5"
+rand = "0.4"
+ndarray = { version = "0.11" }
+time = "0.1"
+colored = "1.6"
```
Comment (kali) [inline cli/Cargo.toml]: i like them ordered by alphabetic order :)

Comment (liautaud) [reply cli/Cargo.toml]: Solved by 84356b46e3d60130885260b7a14606e489d34e96.

File: cli/src/utils.rs [hunk 2]
```diff
@@ -0,0 +1,112 @@
...
+        if s.len() == 0 {
+            let output = model.get_node_by_id(i)?.name.clone();
+            info!("Autodetecting output node: {:?}.", output);
+
+            return Ok(output);
+        }
+    }
+
+    bail!("Impossible to auto-detect output nodes.")
+}
```
[the remark this answers is not in the document]
Comment (liautaud) [reply cli/src/utils.rs]: Solved by 35fcd8d7d8f9077113f3dd2a90ffd5688547844f.

File: cli/src/main.rs [hunk 3]
```diff
@@ -0,0 +1,402 @@
...
+use tfpb::types::DataType;
+use time::PreciseTime;
+
+
+/// The default number of iterations for the profiler.
+const DEFAULT_ITERS: usize = 10000;
+
+
+/// Structure holding the parsed parameters.
+#[allow(dead_code)]
```
[the remark this answers is not in the document]
Comment (liautaud) [reply cli/src/main.rs]: Solved by 8fc2d76b69c9e5ace4cffd12c15a3e4ca777e41d.

File: cli/src/main.rs [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+        path: path.to_string(),
+        graph, tfd_model,
+        inputs, output, size_x, size_y, size_d
+    });
+}
+
+
+/// Handles the `compare` subcommand.
+#[allow(unused_variables)]
+#[cfg(not(feature="tensorflow"))]
```
[the remark this answers is not in the document]
Comment (liautaud) [reply cli/src/main.rs]: Solved by 8fc2d76b69c9e5ace4cffd12c15a3e4ca777e41d.

File: cli/src/main.rs [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+fn handle_compare(params: Parameters) -> Result<()> {
+    bail!("Comparison requires the `tensorflow` feature.")
+}
+
+#[allow(unused_mut)]
+#[allow(unused_imports)]
+#[allow(unused_variables)]
+#[cfg(feature="tensorflow")]
```
[the remark this answers is not in the document]
Comment (liautaud) [reply cli/src/main.rs]: Solved by 8fc2d76b69c9e5ace4cffd12c15a3e4ca777e41d.

File: cli/src/main.rs [hunk 6]
```diff
@@ -0,0 +1,402 @@
...
+    graph: tfpb::graph::GraphDef,
+    tfd_model: tfdeploy::Model,
+
+    #[cfg(feature="tensorflow")]
+    tf_model: conform::tf::Tensorflow,
+
+    inputs: Vec<String>,
+    output: String,
+    size_x: usize,
+    size_y: usize,
```
[the remark this answers is not in the document]
Comment (liautaud) [reply cli/src/main.rs]: Solved by 1bdab8186e20088385aba21cd9037ef64c03548b.

File: cli/src/main.rs [hunk 7]
```diff
@@ -0,0 +1,402 @@
...
+
+        (@arg debug: -d ... "Sets the level of debugging information.")
+
+        (@subcommand compare =>
+            (about: "Compares the output of tfdeploy and tensorflow on randomly generated input."))
+
+        (@subcommand profile =>
+            (about: "Benchmarks tfdeploy on randomly generated input.")
+            (@arg iters: -n [iters]
+                "Sets the number of iterations for the average [default: 10000]."))
```
[the remark this answers is not in the document]
Comment (liautaud) [reply cli/src/main.rs]: Solved by d16e824c06b1703f7d51ed30307cb22f12443649.

File: cli/src/utils.rs [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+                    "Data mismatch: tf={:?}, tfd={:?}",
+                    mtf,
+                    mtfd.borrow()
+                )
+            }
+        }
+    }
+
+    Ok(())
+}
```
[the remark this answers is not in the document]
Comment (liautaud) [reply cli/src/utils.rs]: Solved by 66bd2d263f0823d0f26b65d12e5409ce57384662.

**Pull request merged**